lib/SIL/IR/OperandOwnership.cpp

```diff
@@ -127,559 +122,368 @@ SHOULD_NEVER_VISIT_INST(GetAsyncContinuation)
+  case SILArgumentConvention::Indirect_Inout:
+  case SILArgumentConvention::Indirect_InoutAliasable:
+    llvm_unreachable("Illegal convention for non-address types");
   }
-  llvm_unreachable("unhandled convension");
 }
 
-OwnershipConstraint
-OwnershipConstraintClassifier::visitBeginApplyInst(BeginApplyInst *i) {
+OperandOwnership
+OperandOwnershipClassifier::visitBeginApplyInst(BeginApplyInst *i) {
   return visitFullApply(i);
 }
 
-OwnershipConstraint
-OwnershipConstraintClassifier::visitApplyInst(ApplyInst *i) {
+OperandOwnership OperandOwnershipClassifier::visitApplyInst(ApplyInst *i) {
   return visitFullApply(i);
 }
 
-OwnershipConstraint
-OwnershipConstraintClassifier::visitTryApplyInst(TryApplyInst *i) {
+OperandOwnership
+OperandOwnershipClassifier::visitTryApplyInst(TryApplyInst *i) {
   return visitFullApply(i);
 }
 
-OwnershipConstraint
-OwnershipConstraintClassifier::visitPartialApplyInst(PartialApplyInst *i) {
+OperandOwnership
+OperandOwnershipClassifier::visitPartialApplyInst(PartialApplyInst *i) {
   // partial_apply [stack] does not take ownership of its operands.
-  if (i->isOnStack())
-    return OwnershipConstraint::any();
-
+  if (i->isOnStack()) {
+    return OperandOwnership::InstantaneousUse;
+  }
   // All non-trivial types should be captured.
-  return {OwnershipKind::Owned, UseLifetimeConstraint::LifetimeEnding};
+  return OperandOwnership::ForwardingConsume;
 }
 
-// TODO: FIX THIS
-OwnershipConstraint
-OwnershipConstraintClassifier::visitYieldInst(YieldInst *i) {
-  // Indirect return arguments are address types.
-  //
-  // TODO: Change this to check if this operand is an indirect result
-  if (isAddressOrTrivialType())
-    return OwnershipConstraint::any();
-
-  auto fnType = i->getFunction()->getLoweredFunctionType();
-  auto yieldInfo = fnType->getYields()[getOperandIndex()];
-  switch (yieldInfo.getConvention()) {
-  case ParameterConvention::Indirect_In:
-  case ParameterConvention::Direct_Owned:
-    return visitApplyParameter(OwnershipKind::Owned,
-                               UseLifetimeConstraint::LifetimeEnding);
-  case ParameterConvention::Indirect_In_Constant:
-  case ParameterConvention::Direct_Unowned:
-    // We accept unowned, owned, and guaranteed in unowned positions.
-    return OwnershipConstraint::any();
-  case ParameterConvention::Indirect_In_Guaranteed:
-  case ParameterConvention::Direct_Guaranteed:
-    return visitApplyParameter(OwnershipKind::Guaranteed,
-                               UseLifetimeConstraint::NonLifetimeEnding);
-  // The following conventions should take address types.
-  case ParameterConvention::Indirect_Inout:
-  case ParameterConvention::Indirect_InoutAliasable:
-    llvm_unreachable("Unexpected non-trivial parameter convention.");
-  }
-  llvm_unreachable("unhandled convension");
+// TODO: FIX THIS??
+OperandOwnership OperandOwnershipClassifier::visitYieldInst(YieldInst *) {
+  return getForwardingOperandOwnership();
```

Comment:
@atrick lets talk about this offline.

Reply:
Yield is now handled exactly like all function arguments, which I think is appropriate.